hathor/event/event_manager.py

```diff
@@ -0,0 +1,55 @@
+import time
+
+from structlog import get_logger
+
+from hathor.event.base_event import BaseEvent
+from hathor.event.storage import EventStorage
+from hathor.pubsub import EventArguments, HathorEvents, PubSubManager
+
+logger = get_logger()
+
+
+class EventManager:
+    def __init__(self, event_storage: EventStorage, peer_id: str):
+        self.event_storage = event_storage
+        last_event = event_storage.get_last_event()
+        self.next_event_index = 1 if last_event is None else last_event.id+1
+        self.peer_id = peer_id
+        self.log = logger.new()
+
+    def subscribe(self, pubsub: PubSubManager) -> None:
+        """ Subscribe to defined events for the pubsub received
+        """
+        self.log.error(pubsub)
+        events = [
+            HathorEvents.NETWORK_NEW_TX_ACCEPTED,
+            HathorEvents.NETWORK_NEW_TX_VOIDED,
+            HathorEvents.NETWORK_BEST_BLOCK_FOUND,
+            HathorEvents.NETWORK_ORPHAN_BLOCK_FOUND,
+            HathorEvents.LOAD_STARTED,
+            HathorEvents.LOAD_FINISHED,
+            HathorEvents.REORG_STARTED,
+            HathorEvents.REORG_FINISHED,
+            HathorEvents.TX_METADATA_CHANGED,
+            HathorEvents.BLOCK_METADATA_CHANGED
+        ]
+
+        for event in events:
+            pubsub.subscribe(event, self._persist_event)
+
+    def _persist_event(self, event: HathorEvents, args: EventArguments) -> None:
+        event_data = args.__dict__['event']
+        event_to_store = BaseEvent(
+            id=self.next_event_index,
+            peer_id=self.peer_id,
+            timestamp=int(time.time()),  # Current epoch
```

Comment:
Should we store microseconds to order them correctly?

Reply:
Maybe it makes sense to use floats here, but this isn't an issue for ordering, events are only guaranteed to be sorted by id. If we add any sorting requirement by timestamp, we'd have to make sure an event never has a time lower than the previous event. I'd rather just make the timestamp informative.

Reply:
Fixed.

Reply:
Makes sense to use the id for ordering and the timestamp only for reference. Thanks for the info. Please consider adding this note somewhere, maybe in a comment.

Reply:
Thanks!